openwpm_utils/s3.py

```diff
@@ -89,8 +114,17 @@ def __init__(self, s3_directory, s3_bucket='openwpm-crawls'):
         self._s3_table_loc = "%s/%s/visits/%%s" % (s3_bucket, s3_directory)
         self._content_key = "%s/content/%%s.gz" % s3_directory
         self._s3fs = s3fs.S3FileSystem()
-
-    def read_table(self, table_name, columns=None):
+        self._incomplete_visit_ids = self.read_table(
+            "incomplete_visits", mode="all"
+        ).select("visit_id")
+        crawl_history = self.read_table("crawl_history", mode="all")
+        self._failed_visit_ids = (
+            get_worst_status_per_visit_id(crawl_history)
+            .where(F.col("worst_status") != "ok")
+            .select("visit_id")
+        )
```

Comment:
S3Dataset is mean to be an interface to interact with the data that does not use a spark context. Thus, we can't use any spark functions here and should never be working with spark dataframes here. This is the object you'd use when loading data from S3 in a local jupyter notebook that isn't running spark. It's also an interface that can be used by spark worker nodes (which don't have access to the spark context).